pkg/dns/ibm/private/dnssvcs_provider_test.go

```diff
@@ -35,76 +41,182 @@ func Test_Delete(t *testing.T) {
 			listAllDnsRecordsInputOutput: dnsclient.ListAllDnsRecordsInputOutput{
-				OutputError:      nil,
-				OutputStatusCode: http.StatusOK,
+				OutputError:    nil,
+				OutputResponse: &core.DetailedResponse{StatusCode: http.StatusOK},
+				OutputResult: &dnssvcsv1.ListResourceRecords{
+					ResourceRecords: []dnssvcsv1.ResourceRecord{{
+						ID:    pointer.String("testDelete"),
+						Name:  pointer.String("testDelete"),
+						Rdata: map[string]interface{}{"ip": "11.22.33.44"},
+						Type:  pointer.String(string(iov1.ARecordType)),
+					}},
+				},
 			},
 			deleteDnsRecordInputOutput: dnsclient.DeleteDnsRecordInputOutput{
-				InputId:          "testDelete",
-				OutputError:      errors.New("error in DeleteDnsRecord"),
-				OutputStatusCode: http.StatusRequestTimeout,
+				InputId:        "testDelete",
+				OutputError:    errors.New("error in DeleteDnsRecord"),
+				OutputResponse: &core.DetailedResponse{StatusCode: http.StatusRequestTimeout},
 			},
 			expectErrorContains: "error in DeleteDnsRecord",
 		},
+		{
+			desc:    "list success but mismatching target",
+			DNSName: "testDelete",
+			target:  "11.22.33.44",
+			listAllDnsRecordsInputOutput: dnsclient.ListAllDnsRecordsInputOutput{
+				OutputError:    nil,
+				OutputResponse: &core.DetailedResponse{StatusCode: http.StatusOK},
+				OutputResult: &dnssvcsv1.ListResourceRecords{
+					ResourceRecords: []dnssvcsv1.ResourceRecord{{
+						ID:    pointer.String("testDelete"),
+						Name:  pointer.String("testDelete"),
+						Rdata: map[string]interface{}{"ip": "22.33.44.55"},
+						Type:  pointer.String(string(iov1.ARecordType)),
+					}},
+				},
+			},
+		},
+		{
+			desc:    "list success but record type is nil",
+			DNSName: "testDelete",
+			target:  "11.22.33.44",
+			listAllDnsRecordsInputOutput: dnsclient.ListAllDnsRecordsInputOutput{
+				OutputError:    nil,
+				OutputResponse: &core.DetailedResponse{StatusCode: http.StatusOK},
+				OutputResult: &dnssvcsv1.ListResourceRecords{
+					ResourceRecords: []dnssvcsv1.ResourceRecord{{
+						ID:    pointer.String("testDelete"),
+						Name:  pointer.String("testDelete"),
+						Rdata: map[string]interface{}{"ip": "11.22.33.44"},
+					}},
+				},
+			},
+			expectErrorContains: "delete: failed to get resource type, resourceRecord.Type is nil",
+		},
+		{
+			desc:    "list success but nil results",
+			DNSName: "testDelete",
+			listAllDnsRecordsInputOutput: dnsclient.ListAllDnsRecordsInputOutput{
+				OutputError:    nil,
+				OutputResponse: &core.DetailedResponse{StatusCode: http.StatusOK},
+			},
+			expectErrorContains: "delete: ListResourceRecords returned nil as result",
+		},
+		{
+			desc:    "list success but nil dns record list",
+			DNSName: "testDelete",
+			listAllDnsRecordsInputOutput: dnsclient.ListAllDnsRecordsInputOutput{
+				OutputError:    nil,
+				OutputResponse: &core.DetailedResponse{StatusCode: http.StatusOK},
+				OutputResult:   &dnssvcsv1.ListResourceRecords{},
+			},
+		},
```

Comment:
The looks the same as "create happy path": https://github.com/openshift/cluster-ingress-operator/blob/c5d48ec7eeafdfa58cf005ff50d822228db65ca6/pkg/dns/ibm/private/dnssvcs_provider_test.go#L385-L400

Ah, but "list success but nil dns record list" is missing `target`.  Should it rather be called "list success but _no matching entry_ in dns record list"?

Reply:
Well, I guess the "create happy path" is the same as "no matching entry in dns record list".  In both cases, the list happens to be empty, but in this case, the target is an empty string.  

It's weird that specifying an empty target doesn't cause an error, and it isn't clear that this test was intended to test that scenario.  Given that that's the current behavior, I don't mind having a test case.  However, the test case should be named clearly (something like "create with an empty target"), and we might follow-up with a fix to reject records with empty target and update the test case accordingly to expect an error.

Reply:
I think you misplaced your comment on the delete unit test, but I think you meant to comment on the one for create (same named test cases).

Yea I actually think your first thought is correct, it feels redundant. The target does not impact anything in this test. Right, I was going to say, maybe it provides value by encoding/protecting a specific behavior, but that wasn't the intention.

I'll at least update the name.